llvm/lib/Target/PowerPC/PPCISelLowering.cpp

```diff
@@ -15771,33 +15808,33 @@ SDValue PPCTargetLowering::combineVectorShuffle(ShuffleVectorSDNode *SVN,
     // the value into element zero. Since scalar size of LHS and RHS may differ
     // after isScalarToVec, this should be checked using their own sizes.
     if (SToVLHS) {
-      if (!IsLittleEndian && SToVLHS.getValueType().getScalarSizeInBits() >= 64)
+      int LHSScalarSize = SToVLHS.getValueType().getScalarSizeInBits();
+      if (!IsLittleEndian && LHSScalarSize >= 64)
         return Res;
-      // Set up the values for the shuffle vector fixup.
-      LHSMaxIdx = NumEltsOut / NumEltsIn;
-      SToVLHS = getSToVPermuted(SToVLHS, DAG, Subtarget);
-      if (SToVLHS.getValueType() != LHS.getValueType())
-        SToVLHS = DAG.getBitcast(LHS.getValueType(), SToVLHS);
-      LHS = SToVLHS;
+      LHS = generateSToVPermutedForVecShuffle(
+          LHSScalarSize, ShuffleEltWidth, LHSNumValidElts, LHSFirstElt,
+          LHSLastElt, LHS, SToVLHS, DAG, Subtarget);
     }
     if (SToVRHS) {
-      if (!IsLittleEndian && SToVRHS.getValueType().getScalarSizeInBits() >= 64)
+      int RHSScalarSize = SToVRHS.getValueType().getScalarSizeInBits();
+      if (!IsLittleEndian && RHSScalarSize >= 64)
```

Comment:
I think if we hit run   `LHS = generateSToVPermutedForVecShuffle`
but  `!IsLittleEndian && RHSScalarSize >= 64` is true

it still need to run 
function `isShuffleMaskInRange`

Reply:
Do you mean, do we still need to run `isShuffleMaskInRange()` if we generate a `SCALAR_TO_VECTOR_PERMUTED` for the LHS, but not the RHS side?

I don't think so. I think this is due to the above comment:
```
    // On big endian systems, this only makes sense for element sizes smaller
    // than 64 bits since for 64-bit elements, all instructions already put
    // the value into element zero. Since scalar size of LHS and RHS may differ
    // after isScalarToVec, this should be checked using their own sizes.
```

Reply:
if (!IsLittleEndian && RHSScalarSize >= 64)
        return Res;
        
  the patch still generate LHS, which are not used actually.
  
  can we change the code to
  
  
     ```
 if (SToVLHS) {
      LHSScalarSize = SToVLHS.getValueType().getScalarSizeInBits();
      if (!IsLittleEndian && LHSScalarSize >= 64)
        return Res;
    }

    if (SToVRHS) {
      int RHSScalarSize = SToVRHS.getValueType().getScalarSizeInBits();
      if (!IsLittleEndian && RHSScalarSize >= 64)
        return Res;
    }

    if(LHSScalarSize != 0)
      LHS = generateSToVPermutedForVecShuffle(
          LHSScalarSize, ShuffleEltWidth, LHSNumValidElts, LHSFirstElt,
          LHSLastElt, LHS, SToVLHS, DAG, Subtarget);

   if (RHSScalarSize != 0 )
      RHS = generateSToVPermutedForVecShuffle(
          RHSScalarSize, ShuffleEltWidth, RHSNumValidElts, RHSFirstElt,
          RHSLastElt, RHS, SToVRHS, DAG, Subtarget);

    if (!isShuffleMaskInRange(ShuffV, HalfVec, LHSLastElt, RHSLastElt))
      return Res;
  
```